playbooks/roles/load-balancer-v2/tasks/main.yml

```diff
@@ -154,27 +154,12 @@
     - "stage/*.pem"
   when: inventory_hostname in groups['load-balancer-v2-stage']
 
-- name: Ensure HAProxy picks up conf.d rules 
+- name: Ensure HAProxy picks up conf.d rules
   lineinfile:
     path: /etc/default/haproxy
     line: EXTRAOPTS="-f /etc/haproxy/conf.d"
     create: yes
 
-- name: Set up cron job for the load balancer certificate
-  # To renew periodically. Let's Encrypt certs expire every 3 months.
-  # https://certbot.eff.org/docs/using.html#certbot-command-line-options
-  cron:
-    name: "Renew LB certificate using the certbot client"
-    # We shouldn't need to pass any parameters to "certbot renew", since it
-    # reuses the parameters used when a certificate was issued for the last time.
-    # However, not passing in the port results in letsencrypt complaining about
-    # port 443 being in use.  (Seems to be a bug in certbot.)
-    job: certbot renew --preferred-challenges http-01 --http-01-port 8080 -q
-    hour: "*/12"
-    minute: 42
-    cron_file: letsencrypt-renew
-    user: root
-
```

Comment:
@mtyaka, is it safe to remove this? Iirc, the ACME verification requests on the default HTTP port are all proxied to the {{cert-manager}} server and this custom port `8080` is used for validating the SSL certificate for the `<haproxy hostname>` certificate. Removing this will prevent those certificates from being renewed.

Does the default cron job automatically renew this certificate using a custom port as well? While running `/snap/bin/certbot certificates` lists the certificate for the haproxy host name, I don't see anything in the renewal configuration to indicate that a separate port has to be used for the validation.

Reply:
@lgp171188 

> Iirc, the ACME verification requests on the default HTTP port are all proxied to the {{cert-manager}} server and this custom port 8080 is used for validating the SSL certificate for the <haproxy hostname> certificate.

It used to be like that in the past, but now requests for the haproxy certificates itself are [also proxied via haproxy|https://github.com/open-craft/ansible-playbooks/blob/8b0ba942993a70d255054821ea643aac39199195/playbooks/roles/load-balancer-v2/templates/haproxy.cfg.ctmpl#L51] so that is no longer needed. This cronjob was not working anymore, it was failing with "Problem binding to port 8080: Could not bind to IPv4 or IPv6.. Skipping.".

> While running /snap/bin/certbot certificates lists the certificate for the haproxy host name, I don't see anything in the renewal configuration to indicate that a separate port has to be used for the validation.

I manually removed the port parameters from certificate renewal configurations where they were still present since they're no longer needed as mentioned above. I verified renewals with updated configurations were working with `certbot renew --dry-run. I can also see that the certificates on haproxy-stage-* servers got successfully renewed with the new cronjob yesterday and settings during the night.

Reply:
Thank you for the details @mtyaka 👍🏼